Framework/Algorithms/src/MaxEnt.cpp

```diff
@@ -73,19 +74,19 @@ MatrixWorkspace_sptr removeZeros(const MatrixWorkspace_sptr &ws,
     label->setLabel("Number of Iterations", "");
   }
 
-  if (maxIt == ws->readY(0).size()) {
-    return ws;
-  }
   const size_t nspec = ws->getNumberHistograms();
-  MatrixWorkspace_sptr outWS =
-      WorkspaceFactory::Instance().create(ws, nspec, maxIt, maxIt);
+  if (itCount.size() == 0) {
```

Comment:
It is generally better practice to use `itCount.empty()` instead of checking for a `size==0`.